app/controllers/main_controller.rb

```diff
@@ -130,17 +138,17 @@ def refresh_session
   # same for regular logins and are different on an assume role call.
   # If the login keyword is true then this method also authenticates the real_user
   #
-  def validate_login(user_name, password)
-    if user_name.blank? || password.blank?
+  def validate_login(user_name, password, auth_type: User::AUTHENTICATE_LOCAL)
+    if user_name.blank? || (password.blank? && auth_type == User::AUTHENTICATE_LOCAL)
       flash_now(:error, get_blank_message(user_name, password))
       return false
     end
 
     # No validate file means only remote authentication is allowed
```

Comment:
This comment also needs to be updated

Reply:
@pretendWhale it doesn't seem like you changed this comment. My point here is that the comment is now misleading because the check below isn't just checking for "no validate file".